### PR TITLE
Fix invalid toast message for evaluation error

### LINKED ISSUE #2432: [Bug] Seeing an error toast when none is expected during evaulation
## Description

A toast shows up saying "Unexpected error occurred while evaluating the app" when an action is triggered which leads to some widgets reloading. Before the widget loads with the new data, a toast stating the aforementioned error is displayed.

This particular state (before the new data is put in the widgets) is expected and the error is breaking the user expectations.

Attaching the following video which captures this issue : 
https://www.loom.com/share/b0592774411547c0b679bc45d463839a

### PR DESCRIPTION
**Pull request by hetunandu**
## Description
There was a toast message shown to the user on action execution regarding and unexpected error when evaluating. This was because of bad execution params being sent for evaluation. This PR fixes the issue

Fixes #2432
Fixes: #2429

## Type of change

- Bug fix (non-breaking change which fixes an issue)

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes

### CODE REVIEW

File: app/client/src/sagas/ActionExecutionSagas.ts [hunk 1]
```diff
@@ -281,21 +281,27 @@ const EXECUTION_PARAM_REFERENCE_REGEX = /this.params/g;
...
+  executionParams?: Record<string, any> | string,
 ) {
   if (_.isNil(bindings)) return [];
-  // This might look like a bug, but isn't.
-  // We send in stringified executionParams, but get back an object
-  const evaluatedExecutionParams = yield evaluateDynamicBoundValueSaga(
-    JSON.stringify(executionParams),
-  );
-
+  // We might get execution params as an object or as a string.
```
Comment (ofpiyush) [inline app/client/src/sagas/ActionExecutionSagas.ts:289]: Should we error out here / somewhere before this if we got a string.

Unless we want to process the response, showing a validation error early on might help the user.

**Review by ofpiyush — CHANGES_REQUESTED** [review]

File: app/client/src/workers/evaluation.worker.ts [hunk 2]
```diff
@@ -115,6 +115,7 @@ ctx.addEventListener(
       }
       case EVAL_WORKER_ACTIONS.EVAL_SINGLE: {
         const { binding, dataTree } = requestData;
+        console.log("inside", { dataTree });
```
Comment (ofpiyush) [inline app/client/src/workers/evaluation.worker.ts:118]:
```suggestion
```

**Review by ofpiyush — APPROVED** [review]

**Comment by ofpiyush** [comment]
We should stop showing the toast, as it isn't helping the user discover "where" the error was.

**Comment by hetunandu** [comment]
> We should stop showing the toast, as it isn't helping the user discover "where" the error was.

Removed the toast

**Review by ofpiyush — APPROVED** [review]

**Pull request merged**